### PR TITLE
Add PyCode_GetCode() function

### PR DESCRIPTION
**Pull request by vstinner**

### CODE REVIEW

**Comment by vstinner** [comment]
Wait until https://github.com/python/cpython/pull/92168 is released.

File: docs/api.rst [hunk 1]
```diff
@@ -27,6 +27,12 @@ Latest version of the header file:
 Python 3.11
 -----------
 
+.. c:function:: PyObject* PyCode_GetCode(PyObject *code)
```
Comment (Fidget-Spinner) [inline docs/api.rst:30]: Nit: ``PyCodeObject *code`` not ``PyObject *code`` :)

File: pythoncapi_compat.h [hunk 2]
```diff
@@ -487,6 +487,15 @@ PyFloat_Unpack8(const char *p, int le)
 #endif
 
 
+// gh-92154 added PyCode_GetCode() to Python 3.11.0b1
+#if PY_VERSION_HEX < 0x030B00B1 && !defined(PYPY_VERSION)
```
Comment (Fidget-Spinner) [inline pythoncapi_compat.h:491]: Not sure if it matters, but this code will fail on one version: 3.11a7. In that version, co_code was removed, but there is no PyCode_GetCode.

I don't know what you should do here. Maybe exclude that version?

Comment (erlend-aasland) [reply pythoncapi_compat.h:491]: Perhaps `#error` with an explanatory message.

Comment (vstinner) [reply pythoncapi_compat.h:491]: > I don't know what you should do here. Maybe exclude that version?

Alpha versions of Python 3.11 are not supported: https://github.com/python/pythoncapi-compat/blob/c9764f51fbfd391868e6e81bf0435a445eb25d07/tests/test_pythoncapi_compat.py#L152-L157

It's a deliberate choice to not pollute pythoncapi_compat.h with ugly code to tons of compatibility layers, whereas there are few users of Python alpha versions.

> Perhaps #error with an explanatory message.

What I suggested to users who want to upgrade pythoncapi_compat.h before Python 3.11 beta1 is to copy the file and then remove the functions which cause compilation errors. It's being discussed at: https://github.com/python/pythoncapi-compat/issues/32

File: docs/api.rst (hunk 1 above)
Comment (vstinner) [reply docs/api.rst:30]: Fixed in commit bed155efcc133b0e901857fb4c964e84429087eb.

**Pull request merged**